### PR TITLE
Candy machine post sale mass signer

### PR DESCRIPTION
**Pull request by solserer-labs**
This pull request implements a script that will:

- Get accounts by candy machine creator address
- Get only verified ones
- Get only unverified ones with creator address (keypair address)
- Grab n at a time and batch sign and send transaction

Note that it also adds another build mode `package:linuxb` as building with the --no-bytecode flag didn't work for me. Idk what's the best way to about this, any suggestions?

### CODE REVIEW

File: js/packages/cli/src/cli.ts [hunk 1]
```diff
@@ -583,4 +583,42 @@ program
...
+    '-e, --env <string>',
+    'Solana cluster env name',
+    'devnet', //mainnet-beta, testnet, devnet
+  )
+  .option(
+    '-k, --keypair <path>',
+    `Solana wallet location`,
+    '',
+  )
+  .option('-cndy, --candy-address <string>', 'Candy machine address', '')
```
Comment (WrRaThY) [inline js/packages/cli/src/cli.ts:598]: address may be taken from cache

File: js/packages/cli/src/cli.ts [hunk 2]
```diff
@@ -583,4 +583,42 @@ program
...
+      console.log("Batch size needs to be an integer!")
+      return;
+    }
+    const walletKeyPair = loadWalletKey(keypair);
+    const anchorProgram = await loadAnchorProgram(walletKeyPair, env);
+    console.log("Creator pubkey: ", walletKeyPair.publicKey.toBase58())
+    console.log("Environment: ", env)
+    console.log("Candy machine address: ", candyAddress)
+    console.log("Batch Size: ", bVal)
+    await signAllMetadataFromCandyMachine(anchorProgram.provider.connection, walletKeyPair, candyAddress, bVal)
```
Comment (WrRaThY) [inline js/packages/cli/src/cli.ts:621]: providing anchorProgram would be cleaner
rename bVal batchSizeInt or batchSizeParsed or smth like that

File: js/packages/cli/src/signer.ts [hunk 3]
```diff
@@ -0,0 +1,435 @@
...
+ Get only unverified ones with creator address
+ Grab n at a time and batch sign and send transaction
+
+ PS: Don't sign candy machine addresses that you do not know about. Signing verifies your participation.
+*/
+const MAX_NAME_LENGTH = 32;
+const MAX_URI_LENGTH = 200;
+const MAX_SYMBOL_LENGTH = 10;
+const MAX_CREATOR_LEN = 32 + 1 + 1;
+const METADATA_PROGRAM_ID = "metaqbxxUerdq28cj1RbAWkYQm3ybzjb6a8bt518x1s"
```
Comment (WrRaThY) [inline js/packages/cli/src/signer.ts:17]: move them to constants (METADATA_PROGRAM_ID  is already there)

File: js/packages/cli/src/signer.ts [hunk 4]
```diff
@@ -0,0 +1,435 @@
...
+}
+
+class Creator {
+  address: string;
+  verified: boolean;
+  share: number;
+
+  constructor(args: {
+    address: string;
+    verified: boolean;
```
Comment (WrRaThY) [inline js/packages/cli/src/signer.ts:103]: all of those should go to `types.ts`

**Review by WrRaThY — COMMENTED** [review]
a couple of things that could be done better, but you might just want to merge your changes here:
https://github.com/metaplex-foundation/metaplex/pull/337
I tried to tackle that before just couldn't figure out how to find all those account (which you did), I just think that the structure I proposed those is a bit clearer

File: js/packages/cli/src/cli.ts (hunk 1 above)
Comment (solserer-labs) [reply js/packages/cli/src/cli.ts:599]: afaik address isn't being added to the .cache file only the config address.

i added a json file that will store all the candy machines you make from the cli inside the .cache folder

File: js/packages/cli/src/cli.ts (hunk 2 above)
Comment (solserer-labs) [reply js/packages/cli/src/cli.ts:627]: i've only now started to look into anchor

**Comment by solserer-labs** [comment]
ok merged with latest master and tested it seems to work

**Comment by solserer-labs** [comment]
moved the signer to commands and using cache for candy machine address now

**Pull request merged**